### DISCUSSION TITLE
Filament 3, custom classes

### DISCUSSION THREAD

**Discussion by hamzadevtech1** [filamentphp/filament#10161] [Help]
I have upgraded my application from filament 2 to 3 version. My configurations are working fine. The only problem I have is, I have given a viteTheme like this

 ->viteTheme('resources/css/filament.css', 'build')
 
 and my filament.css file has tailwind included with all default colors, like primary, secondary, danger etc.
 But Filament is applying this class everywhere e.g bg-custom-500
 
 and because of this custom class, it does not match my theme. Is there a way to tell Filament to consider my primary tailwind classes as primary, instead of this custom class ?

**Comment by devhoussam1998** [comment]
did you check 

https://filamentphp.com/docs/3.x/support/colors

**Reply by MickProPay** [reply]
Yes we checked that link but the problem the new setup is VERY limiting the old method we had a whole css file that effect scrollbar and component colour and a lot more. So we want to make use of the vite call like in version 2...

Also they create a custom css file that takes our primary colour and then even applies a lighter version of the primary colour...

**Reply by devhoussam1998** [reply]
I think you need to check this https://filamentphp.com/docs/3.x/support/render-hooks#registering-render-hooks

**Reply by hamzadevtech1** [reply]
We checked that. Actually we want to use the viteTheme for our filament and use the colors from the tailwind config file... But Filament docs does not have anything on configuring viteTheme in appPanelProvider. If you could please provide the link for that then that would be great

**Reply by hamzadevtech1** [reply]
Also, How Filament Maps the colors with bg-custom-500 ? like whatever color we provide, it shows the custom class and maps that color to it. Can't be define our own primary custom color ?

**Comment by masterix21** [comment]
@hamzadevtech1, add to your tailwind.config.js the custom color like this:
```js
theme: {
        extend: {
            colors: {
                danger: colors.red,
                gray: colors.zinc,
                info: colors.blue,
                primary: colors.indigo,
                secondary: colors.gray,
                success: colors.green,
                warning: colors.orange,
                custom: { // it :-)
                    50: 'rgba(var(--c-50), <alpha-value>)',
                    100: 'rgba(var(--c-100), <alpha-value>)',
                    200: 'rgba(var(--c-200), <alpha-value>)',
                    300: 'rgba(var(--c-300), <alpha-value>)',
                    400: 'rgba(var(--c-400), <alpha-value>)',
                    500: 'rgba(var(--c-500), <alpha-value>)',
                    600: 'rgba(var(--c-600), <alpha-value>)',
                    700: 'rgba(var(--c-700), <alpha-value>)',
                    800: 'rgba(var(--c-800), <alpha-value>)',
                    900: 'rgba(var(--c-900), <alpha-value>)',
                    950: 'rgba(var(--c-950), <alpha-value>)',
                },
            },
        },
    },
```
